src/Driver/PDO/Connection.php

```diff
@@ -80,6 +80,17 @@ public function query(string $sql): ResultInterface
      */
     public function quote($value, $type = ParameterType::STRING)
     {
+        assert(in_array($type, [
+            PDO::PARAM_NULL,
+            PDO::PARAM_INT,
+            PDO::PARAM_STR,
+            PDO::PARAM_LOB,
+            PDO::PARAM_STMT,
+            PDO::PARAM_BOOL,
+            PDO::PARAM_STR_NATL,
+            PDO::PARAM_INPUT_OUTPUT,
+        ], true));
```

Comment:
I'm not sure that we can/should guarantee that.

Reply:
You are right.

I have to override the `$type`.